test/test.js

```diff
@@ -86,9 +86,12 @@ function testInput(input) {
       });
 
       it('should generate valid Swagger 2.0 document', function() {
-        SwaggerTools.specs.v2.validate(converted, function(validationErrors) {
-          expect(validationErrors).to.be.undefined;
-        });
+        function validateCallback(validationErrors, validationResults) {
+          var errors = [].concat(validationErrors || [])
```

Comment:
`validationErrors` is not an array. it should fail as soon as `validationErrors` is truthy 

`validationResults.errors` should be an empty array to pass. 

You don't have to concat `validationErrors` and `validationResults`